core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -3003,8 +3006,62 @@ private String escapeLdapNameForFilter(LdapName ldn){
      * @param dnPartial
      * @return
      */
+    private String escapeSpecialCharactersForFilterWithStarAsRegex(String dnPartial){
```

Comment:
Thanks for pointing this. Fixed with https://github.com/wso2/carbon-kernel/pull/774